File: medico/Domain/Medico.Domain/Models/Allergy.cs
using System;

namespace Medico.Domain.Models
{
    public class Allergy : Entity
    {
        public string Reaction { get; set; }

        public string Medication { get; set; }

        public string Notes { get; set; }

        public Guid? MedicationNameId { get; set; }

        public MedicationName MedicationName { get; set; }

        public Guid? MedicationClassId { get; set; }

        public MedicationClass MedicationClass { get; set; }

        public Guid PatientId { get; set; }

        public Patient Patient { get; set; }

        public DateTime CreateDate { get; set; }
    }
}

File: PatientPortal/src/app/patientChart/models/allergy.ts
export class Allergy {
    id: string;
    reaction: string;
    medication: string;
    medicationNameId: string;
    medicationClassId: string;
    notes: string;
    patientId: string;
    createDate: any;

    constructor() {
        this.createDate = new Date();    
    }
}